src/crypto/rfc8439.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <crypto/rfc8439.h>
+
+#include <crypto/chacha20.h>
+#include <crypto/common.h>
+
+#include <cstring>
+
+#ifndef RFC8439_TIMINGSAFE_BCMP
+#define RFC8439_TIMINGSAFE_BCMP
+
+int rfc8439_timingsafe_bcmp(const unsigned char* b1, const unsigned char* b2, size_t n)
```

Comment:
In 7a9d2fb8cf11eef8b6bd87ad1066df374bf3b620:
It is fairly awkward to introduce a second copy of `timingsafe_bcmp` (one already in `chacha_poly_aead`), with a different name, and broken #ifdef logic, just to then rename it when you delete the original `timingsafe_bcmp` in a later commit. Although I guess re-using/extracting `timingsafe_bcmp` early from the to-be-deleted file is also a bit awkward.